patches/v8/0017-Add-v8-Date-ToUTCString-and-v8-Date-Parse-APIs.patch

```diff
@@ -0,0 +1,124 @@
+From 3525cbcde49dbf58e8ec68a957eb4f3ba5412715 Mon Sep 17 00:00:00 2001
+From: James M Snell <jasnell@gmail.com>
+Date: Sat, 24 Feb 2024 06:12:13 -0800
+Subject: Add v8::Date::ToUTCString and v8::Date::Parse APIs
+
+---
+ include/v8-date.h |  5 +++-
+ src/api/api.cc    | 66 +++++++++++++++++++++++++++++++++++++++++++++++
+ 2 files changed, 70 insertions(+), 1 deletion(-)
+
+diff --git a/include/v8-date.h b/include/v8-date.h
+index 8d82ccc9ea60bbeed6d1903dac3638a921a4a7e3..4f014927b0bf0c617879c46d451ecf5d67fde26f 100644
+--- a/include/v8-date.h
++++ b/include/v8-date.h
+@@ -20,7 +20,8 @@ class V8_EXPORT Date : public Object {
+  public:
+   static V8_WARN_UNUSED_RESULT MaybeLocal<Value> New(Local<Context> context,
+                                                      double time);
+-
++  static V8_WARN_UNUSED_RESULT MaybeLocal<Value> Parse(Local<Context> context,
++                                                       Local<String> date_string);
+   /**
+    * A specialization of Value::NumberValue that is more efficient
+    * because we know the structure of this object.
+@@ -32,6 +33,8 @@ class V8_EXPORT Date : public Object {
+    */
+   v8::Local<v8::String> ToISOString() const;
+ 
++  v8::Local<v8::String> ToUTCString() const;
++
+   V8_INLINE static Date* Cast(Value* value) {
+ #ifdef V8_ENABLE_CHECKS
+     CheckCast(value);
+diff --git a/src/api/api.cc b/src/api/api.cc
+index c0bfc7a16f8f5be37e8c64ec7d2016409fead2db..678aed75a25f73d7fb34afc0681dd9b6edee4a1f 100644
+--- a/src/api/api.cc
++++ b/src/api/api.cc
+@@ -46,6 +46,7 @@
+ #include "src/common/globals.h"
+ #include "src/compiler-dispatcher/lazy-compile-dispatcher.h"
+ #include "src/date/date.h"
++#include "src/date/dateparser.h"
+ #include "src/debug/debug.h"
+ #include "src/deoptimizer/deoptimizer.h"
+ #include "src/execution/embedder-state.h"
+@@ -7917,6 +7918,57 @@ MaybeLocal<v8::Value> v8::Date::New(Local<Context> context, double time) {
+   RETURN_ESCAPED(result);
+ }
+ 
++namespace {
++// This is copied from builtins-date.cc... a cleaner solution would be to make
```

Comment:
I don't think we should be copying this. This code could easily change over time, either to fix bugs or because the internal interfaces it calls have changed. This doesn't seem worth it to maintain this just to avoid having to call the method through JS.